app/javascript/react/screens/App/Settings/__tests__/index.test.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import thunk from 'redux-thunk';
+import promiseMiddleware from 'redux-promise-middleware';
+import { createStore, applyMiddleware, combineReducers } from 'redux';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import { reducer as formReducer } from 'redux-form';
+
+import Settings from '../Settings';
+import SettingsContainer, { reducers } from '../index';
+import { initialState } from '../SettingsReducer';
+import { servers, settings } from '../settings.fixures';
+
+import { mockRequest } from '../../../../../common/mockRequests';
+
+describe('Settings integration test', () => {
```

Comment:
I'd like to discuss not continuing with these index.js integration tests. We can keep the ones that are already in place, but it's been my experience that they are brittle, and failures often are false positives that stem from test environment configuration issues, not actual issues with code/changes.

For example, in the edit mapping PR, the test for MappingWizard needed configuration updates to pass, despite there not having been any actual regressions.

While I think it's not a bad thing to have checks in place for our overall system architecture/configuration, there should probably only be one test at the highest level, not for every container component.

Thoughts?

Reply:
I think this is reasonable. All we're testing in these integration tests is that the `mapStateToProps` function applies a mocked store's state to the component as specified, and in the case where we just return some keys from the store, the test is a bit of a tautology. I think they could be useful if we had complex logic in `mapStateToProps`, but we should instead favor moving that logic to selectors/helpers, and testing those helpers separately.

I'll remove this one.